src/index.js

```diff
@@ -297,14 +297,31 @@ Multiaddr.prototype.equals = function equals (addr) {
  * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
  */
 Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  if (!this.isThinWaistAddress()) {
-    throw new Error('Multiaddr must be "thin waist" address for nodeAddress.')
-  }
-
   const codes = this.protoCodes()
   const parts = this.toString().split('/').slice(1)
+
+  if (parts.length < 4) {
+    throw new Error('Multiaddr must be a valid address for nodeAddress.')
+  }
+
+  let family
+
+  switch (codes[0]) {
+    case 41:
+      family = 'IPv6'
+      break
+    case 54:
+      family = 'DNSv4'
+      break
+    case 55:
+      family = 'DNSv6'
+      break
+    default:
+      family = 'IPv4'
+  }
+
   return {
-    family: (codes[0] === 41) ? 'IPv6' : 'IPv4',
+    family: family,
```

Comment:
This should only be set to either the number `4` or `6`. See https://nodejs.org/api/http.html#http_http_request_options_callback

Also, adding a test case with using whatever `nodeAddress` returns can be passed to `http.request` would be great, as that would have caught this.

Reply:
I didn't know that. I will modify it, setting `IPv6` for ip6, as well as dns6.